tools/src/main/scala/scala/scalanative/codegen/llvm/MetadataCodeGen.scala

```diff
@@ -68,9 +71,24 @@ trait MetadataCodeGen { self: AbstractCodeGen =>
       producer = Constants.PRODUCER,
       isOptimized = attrs.opt == nir.Attr.DidOpt
     )
+    // unmangle method name
+    // see: https://scala-native.org/en/stable/contrib/mangling.html
+    val linkageName = mangled(name)
+    val defnName = if (linkageName.startsWith("_S")) linkageName.drop(2) else linkageName
```

Comment:
I'm not sure, but I think that on Windows there can be just a 'S` prefix, however we should probably check that. We had same special handling for genreating stacktraces on Windows, but I'm not sure if it applies to debug metadata

https://github.com/scala-native/scala-native/blob/4e9ca7d0b478765ee81e3a698e33d037ac0b407a/nativelib/src/main/scala/scala/scalanative/runtime/SymbolFormatter.scala#L31-L49

Reply:
Oh, thanks, I'll check that.

Reply:
I haven't confirmed on my Windows machine (since I'm struggling to setup dev environment on Win), it looks like we definitely have `_S` prefix for mangled name at least on NIR -> LLVM IR https://github.com/scala-native/scala-native/blob/439188535a050ae87f49aa2913cd9f69757a4196/tools/src/main/scala/scala/scalanative/codegen/llvm/AbstractCodeGen.scala#L594-L595

Maybe Windows linker or Windbg's `get_proc_name` drops the `_` prefix? https://github.com/scala-native/scala-native/pull/2287/commits/9f7f24c607bc10ff6e953886ccd4638fca3e33a1

Reply:
Maybe, that's possible. Let's merge this change, and potentially fix it later if we'd get a bug report